src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -935,6 +924,18 @@ zocl_xclbin_read_axlf(struct drm_zocl_dev *zdev, struct drm_zocl_axlf *axlf_obj,
 		goto out0;
 	}
 
+	write_unlock(&zdev->attr_rwlock);
+
+	if (kds_mode == 1) {
+		subdev_destroy_cu(zdev);
+		ret = zocl_create_cu(zdev);
+		if (ret)
+			goto out0;
+		ret = zocl_kds_update(zdev);
+		if (ret)
+			goto out0;
+	}
```

Comment:
Please leave a comment here to explain why this code block should be move out from the write lock region. Then others would not make same mistake and potentially can save a lot of time on debugging.

Reply:
Added a comments. Please check